[Tan Boon Ji] iP

tanboonji opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

tanboonji committed:
Implement greet, echo and exit

tanboonji committed:
Implement add and list

tanboonji committed:
Implement mark as done and Task class

tanboonji committed:
Implement to-dos, events and deadlines

tanboonji committed:
Abstract input processing, list task and exit into individual methods

tanboonji committed:
Fix description for to-do

tanboonji committed:
Update message output for tasks

tanboonji committed:
Implement automated text UI testing

tanboonji committed:
Add error handling for incorrect inputs

tanboonji committed:
Implement task deletion

tanboonji committed:
Update message output for commands and errors

tanboonji committed:
Implement enum for commands

tanboonji committed:
Implement loading and saving of data from and to file

tanboonji committed:
Implement LocalDateTime for Deadline and Event class

The datetime stored in Deadline and Event class are in String.

Let's,
- Modify datetime in Deadline and Event class to LocalDateTime
- Parse input for datetime from String to LocalDateTime
- Add DateParser class to parse datetime in String to LocalDateTime

tanboonji committed:
Merge branch 'branch-Level-7'

tanboonji committed:
Merge branch 'branch-Level-8'

tanboonji committed:
Refractor code to use more OOP

Code logic is lumped inside Duke.

Seperate logic from Duke class into multiple classes to adhere to OOP paradigm and concepts.

Let's,
- Add Ui class
- Add Storage class
- Add TaskList class
- Add seperate Command classes

tanboonji committed:
Add InvalidCommand class

tanboonji committed:
Fix pattern matching and matcher grouping errors for commands

tanboonji committed:
Fix pattern matching for todo command

tanboonji committed:
Fix unexpected NumberFormatException error in DoneCommand class

tanboonji committed:
Fix unexpected NumberFormatException error in DeleteCommand class

tanboonji committed:
Fix parsing error caused by input string splitting in DateParser class

tanboonji committed:
Add developer tests for CommandParser and DateParser class

tanboonji committed:
Add JavaDocs comments

tanboonji committed:
Tweak code to comply with coding standard

tanboonji committed:
Implement find task function

tanboonji committed:
Merge branch 'branch-A-JavaDoc'

tanboonji committed:
Merge branch 'branch-A-CodingStandard'

tanboonji committed:
Merge branch 'branch-Level-9'

DrWala commented on src/main/java/ToDoCommand.java:
```diff
@@ -0,0 +1,39 @@
+        taskList.addTask(newTask);
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("Got it. I've added this task:\n\t").append(newTask).append("\nNow you have ")
```
It might assist readability to break the `.append` into multiple lines since they are new lines, and it makes it clear what is being appended to the `StringBuilder`

DrWala commented on src/main/java/DeadlineCommand.java:
```diff
@@ -0,0 +1,53 @@
+        taskList.addTask(newTask);
+
+        StringBuilder builder = new StringBuilder();
+        builder.append("Got it. I've added this task:\n\t").append(newTask).append("\nNow you have ")
```
It might assist readability to break the `.append` into multiple lines since they are new lines, and it makes it clear what is being appended to the `StringBuilder`.

This specific style issue exists in a few other places too.

DrWala commented on src/main/java/HelpCommand.java:
```diff
@@ -0,0 +1,28 @@
+
+    public static final String COMMAND = "help";
+    public static final String ERROR_MESSAGE = "☹ Sorry, please enter a valid command.\n";
+    public static final String COMMAND_LIST = "\tCommands available:\n\t\t" +
```
Perhaps the same breaking into multiple lines argument goes here too to improve readability, and ease of maintenance should the order be changed or new commands be added.

DrWala reviewed (commented):
LGTM, just a few nits to fix.

tanboonji committed:
Merge branch 'add-gradle-support'

tanboonji committed:
Refactor code for constant Strings and StringBuilder .append() methods

tanboonji committed:
Refactor code into packages

tanboonji committed:
Configure gradle for running checkstyle, tests and creating JAR files

tanboonji committed:
Refactor code detected by checkstyle that violates coding standards

tanboonji committed:
Refactor test code into packages and to comply with coding standards

tanboonji committed:
Implement GUI for Duke

tanboonji committed:
Merge branch 'branch-Level-10'

andrea-twl commented on src/main/java/com/tanboonji/duke/Duke.java:
```diff
@@ -0,0 +1,98 @@
+        ui.greet();
+    }
+
+    private boolean processInput(String input) {
```
Perhaps this method could be named in a way that makes it clearer that it returns a boolean?

andrea-twl reviewed (approved):
Looks good overall!

tanboonji committed:
Update application main class in build.gradle

tanboonji committed:
Update code in MainWindow to comply with checkstyle

tanboonji committed:
Add assertion to ensure task list is not null

Task list is assumed to be either loaded with tasks from saved data or empty if saved data is not found.

Add assertion to esure task list is either of the two mentioned cases above and is not pointing to a null value.
A task list pointing to a null value will cause the program to run into an uncatched NullPointerException.

tanboonji committed:
Merge pull request #1 from tanboonji/A-Assertions

Add assertion to ensure task list is not null

tanboonji committed:
Add title to UI and set UI to non-resizable

tanboonji committed:
Refactor code for Main class to improve code quality

tanboonji committed:
Add JavaDocs for Duke and Storage class

tanboonji committed:
Update data file path from 'duke.txt' to 'duke.data'

tanboonji committed:
Delete dead Ui class and dead code in Duke class

tanboonji committed:
Update JavaDocs for Storage class

tanboonji committed:
Change the declaration order of local variables in Duke class

tanboonji committed:
Add default class constructor for Duke class

tanboonji committed:
Add JavaDocs for command classes

tanboonji committed:
Update JavaDocs for a few classes

tanboonji committed:
Refactor code for command classes to improve code quality

tanboonji committed:
Update JavaDocs for a few classes

tanboonji committed:
Change the usage of HelpCommand and InvalidCommand class

tanboonji committed:
Add JavaDocs for a few classes

tanboonji committed:
Add JavaDocs for DialogBox and MainWindow class

tanboonji committed:
Fix long messages getting trimmed in UI

tanboonji committed:
Fix application not closing after ByeCommand

tanboonji committed:
Refactor code to improve code quality

tanboonji committed:
Merge branch 'master' into A-CodeQuality

tanboonji committed:
Merge pull request #2 from tanboonji/A-CodeQuality

Improve code quality and add JavaDocs

tanboonji committed:
Remove Javadocs checkstyle suppression

tanboonji committed:
Fix string alignment and missing special characters issues on chatbot

tanboonji committed:
Add welcome message to chatbot

tanboonji committed:
Add new custom exceptions that extends from DukeException

tanboonji committed:
Remove dead InvalidCommand class and all usages of the class

tanboonji committed:
Implement alias feature

tanboonji committed:
Add AliasMap class to handle alias hash map

tanboonji committed:
Move list task logic from ListCommand class to TaskList class

tanboonji committed:
Implement list alias and delete alias feature

tanboonji committed:
Merge pull request #4 from tanboonji/branch-C-FriendlierSyntax

Implement friendlier syntax by adding alias feature

tanboonji committed:
Update string response messages for a few command classes

tanboonji committed:
Update code and logic to utilise TaskList and AliasMap toString()

tanboonji committed:
Fix checkstyle issues in some classes

tanboonji committed:
Update duke exit/shutdown logic

tanboonji committed:
Update the GUI to make it more polished

tanboonji committed:
Update the GUI to show seperate icon for error message

tanboonji committed:
Update application name from Duke to Jhin

tanboonji committed:
Remove unused images

tanboonji committed:
Update success message string in AliasCommand class

tanboonji committed:
Add force shutdown when an error occurs while loading saved data

tanboonji committed:
Add JavaDocs for throwed JhinException in some methods

tanboonji committed:
Update force shutdown to prevent it from getting called more than once

tanboonji committed:
Update FindCommand to accept empty string search

tanboonji committed:
Update DateParser to accept more date time formats and usage of trim()

tanboonji committed:
Set up Continuous Integration (CI) using GitHub Actions

tanboonji committed:
Remove unused imports in FindCommand class

tanboonji committed:
Update test cases in JUnit

tanboonji committed:
Update error message string in ToDoCommand class

tanboonji committed:
Update application icon

tanboonji committed:
Update test cases in JUnit

tanboonji committed:
Add Ui.png for iP product showcase

tanboonji committed:
Set theme jekyll-theme-cayman

tanboonji committed:
Update user icon image name

tanboonji committed:
Abstract some code logic into separate methods in MainWindow class

tanboonji committed:
Add CommandType enum and update CommandParser class to use the enum

tanboonji committed:
Update response strings from Jhin and some constant variable names

tanboonji committed:
Update alias command to 'alias <alias> <command>' format

tanboonji committed:
Update JavaDocs for a few methods

tanboonji committed:
Update usage of .trim() method in command and parser classes

tanboonji committed:
Add command regex matcher for DeleteCommand and DoneCommand

tanboonji committed:
Update DoneCommand to check for marking an already done task

tanboonji committed:
Update DateParser to accept more date time formats

tanboonji committed:
Update single angle quotation marks to braces for optional arguments

tanboonji committed:
Add user guide

tanboonji committed:
Update string responses for Ui in command classes and AliasMap

tanboonji committed:
Fix uncaught InvalidCommandArgumentException when calling isTaskDone()

tanboonji committed:
Update Ui size

tanboonji committed:
Update test cases in JUnit

tanboonji committed:
Update user guide

tanboonji committed:
Update user guide

tanboonji committed:
Update user guide

tanboonji committed:
Fix numbering issue in user guide

tanboonji committed:
Update acknowledgement link and a small grammer mistake in user guide

tanboonji committed:
Update gradle to enforce java 11 compatability

tanboonji committed:
Fix command regex matcher to correctly match symbols such as '-' and '!'

tanboonji committed:
Update test cases in JUnit to test for negative integer values
